### ISSUE TITLE
Sorting in the dashboard

### ISSUE THREAD

**Issue by Der-Canonier** [NotePlan/plugins#511]
I work a lot with time blocks. The time blocks in my dashboard are pretty jumbled up.

It would be great if there was a way to sort the "Today" and "Today" (scheduled) section by time. 

z. For example, by the earliest time like in the "Hashtag" section?
![Dashboard sort](https://github.com/NotePlan/plugins/assets/15310331/ad391aa3-4906-4486-aa14-b01c9bb9396e)

**Comment by jgclark** [comment]
I don't use time blocks at all.

I get the idea, but it's one thing to work with a datestamp on every note (as the hashtag uses); it's another thing to use time blocks that can vary in specification on many lines within a note.

@dwertheimer, you probably know time blocks as well as anyone. Do you think there's enough standardisation to make this request feasible?

**Comment by dwertheimer** [comment]
@jgclark I think it should be possible. I don't know how many edge cases there might be. You could approach it like the following (doing this off the top of my head, have not tested it):

Run all the paragraph.content through a sort function that sorts by time.
```js
function sortByTime(a:any, b:any) {
 aTime = Calendar.parseDateText(a.content)
 bTime = Calendar.parseDateText(b.content)
 // determine what you want to do with items which don't have a time (do they go at top or bottom?)
 if (!aTime?.length) // do something
 if (!bTime?.length) // do something
 return aTime[0].start - bTime[0].start // maybe reverse this depending on how you want them ordered
}

const timeSortedParas = allParagraphs.sort(sortByTime) // assuming allParagraphs are NP TParagraph objects
```

**Comment by jgclark** [comment]
Released in 0.8.2. I had to make more detailed assumptions about what you wanted, @Der-Canonier. Please re-open if you find issues with it.

**Comment by Der-Canonier** [comment]
@jgclark 👏🏼 This looks very good for the "Today" section.
One small thing: Task and checklist entries without time blocks are between those with time blocks. 
It might be worth considering whether they should be displayed at the beginning or the end of the list.

Is it a lot of effort to implement the same principle for the "Today" (scheduled) section?

I would find that very cool 😊

**Comment by jgclark** [comment]
> @jgclark 👏🏼 This looks very good for the "Today" section. 

Glad that this is doing what you hoped for ...

> One small thing: Task and checklist entries without time blocks are between those with time blocks. It might be worth considering whether they should be displayed at the beginning or the end of the list.

... but this isn't intended. I didn't see this in my testing. Can you please give me a sample set of items where you're seeing this?

> Is it a lot of effort to implement the same principle for the "Today" (scheduled) section? 
> I would find that very cool 😊

I didn't apply it to this section as I didn't imagine people would do scheduling-within-the-day in project notes, but only in daily notes. So are you really using time blocks within project and daily notes? If so, how do you stop duplication?

**Comment by Der-Canonier** [comment]
> > One small thing: Task and checklist entries without time blocks are between those with time blocks. It might be worth considering whether they should be displayed at the beginning or the end of the list.
> 
> ... but this isn't intended. I didn't see this in my testing. Can you please give me a sample set of items where you're seeing this?

Here is a screenshot of the first question.
I will add the 2nd part for the Scheduled-Today-Section later.

The task or checklist without time blocks are in no particular order under my headings
![Dashboard](https://github.com/NotePlan/plugins/assets/15310331/4db00ee6-eb08-444b-b9b9-2eb270cc0149)
![Noteplan](https://github.com/NotePlan/plugins/assets/15310331/0184b9a8-a121-49fb-abde-5fe49f41467c)

**Comment by Der-Canonier** [comment]
> > Is it a lot of effort to implement the same principle for the "Today" (scheduled) section?
> > I would find that very cool 😊
> 
> I didn't apply it to this section as I didn't imagine people would do scheduling-within-the-day in project notes, but only in daily notes. So are you really using time blocks within project and daily notes? If so, how do you stop duplication?

In fact, I have quite a few project notes with time blocks because I have many notes organized in different folders and subfolders.

- If I want/need to carry out certain tasks at certain times in customer projects
- Because of other appointments or because I can reach the customer or partner at a certain time
- or because I have to do certain things before the start of an appointment.

To name just three examples.

That's why - at least for me - sorting in the dashboard within the scheduled task would be extremely important.

Perhaps the sorting could also be switched on and off via the settings. 

Translated with www.DeepL.com/Translator (free version)

**Comment by Der-Canonier** [comment]
By the way, I have no problem with duplicates.
It's not even clear to me what would cause them

**Comment by jgclark** [comment]
Thanks, that's very clear.  I have added support for this, and will be
included in v0.8.3 along with some other things.

**Comment by Der-Canonier** [comment]
I'm looking forward to the update - thank you very much 🤗
